tests/lib/rules/prefer-const.js

```diff
@@ -56,6 +56,12 @@ ruleTester.run("prefer-const", rule, {
         "let a; function foo() { if (a) {} a = bar(); }",
         "let a; function foo() { a = a || bar(); baz(a); }",
         "let a; function foo() { bar(++a); }",
+        "let predicate; [typeNode.returnType, predicate] = foo();",
+        "let predicate; let rest; [typeNode.returnType, predicate, ...rest] = foo();",
```

Comment:
That's a good point @not-an-aardvark 

In that case, would it make sense to say that the example in #8308 is actually behaving as expected? If `destructuring: "any"` is on by default, it should report that predicate is not reassigned then.